### PR TITLE
Demonstration of ExternalProject and new packaging system.

### PR DESCRIPTION
**Pull request by dbrnz**
This builds zlib using CMake's External Project module (#1161) and also uses the new packaging system (#1106) to create and/or download the resulting library and header files.

The PR also includes a new plugin, SampleExternal, that was developed for prototyping ideas, although it hasn't since been updated to reflect recent changes to the packaging code and so possibly should be removed -- the zlib plugin now provides a fully working example.

### CODE REVIEW

**Comment by agarny** [comment]
@dbrnz, I have started reviewing your PR. It's mainly about style at this stage, but if it's not you who does the changes, it will eventually be me, so... since I am still busy with your PMR Workspaces PR... :)

Otherwise, I have to head back home, so I will finish this tomorrow.

File: src/plugins/sample/SampleExternal/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,117 @@
+PROJECT(SampleExternalPlugin)
```
Comment (agarny) [inline src/plugins/sample/SampleExternal/CMakeLists.txt]: Use uppercase for CMake-related commands.

File: .gitignore [hunk 2]
```diff
@@ -5,3 +5,4 @@
 /src/3rdparty/*/bin/
 /src/3rdparty/*/lib/
 /src/plugins/*/*/bin/
+/src/plugins/*/*/external/
```
Comment (agarny) [inline .gitignore]: Delete, I believe.

**Comment by dbrnz** [comment]
@agarny, I was looking for feedback on my general approach not a detailed critique of what was a work-in-progress... Anyway things have evolved as I worked on packaging Python and this really needs to be a new PR -- something to sort out when you are in Auckland.

**Comment by agarny** [comment]
This being a PR, I took it as being ready for merging into the `master` branch. For feedback on your general approach, I would have expected to be directed to a branch of yours. Anyway, this is something we should indeed look into when I am next in Auckland.

**Comment by agarny** [comment]
Closing this PR since it was 'only' meant as a way to demonstrate the use of `ExternalProject` and of @dbrnz' new packaging system, which has now changed compared with this PR.

**Pull request closed without merging**